portfolio/src/main/java/com/whz/portfolio/infrastructure/persistence/EventAdapter.java

```diff
@@ -0,0 +1,35 @@
+package com.whz.portfolio.infrastructure.persistence;
+
+import com.whz.portfolio.model.portfolio.PortfolioCreated;
+
+import io.vlingo.common.serialization.JsonSerialization;
+import io.vlingo.symbio.BaseEntry.TextEntry;
+import io.vlingo.symbio.EntryAdapter;
+import io.vlingo.symbio.Metadata;
+import io.vlingo.symbio.Source;
+
+public final class EventAdapter<T extends Source<?>> implements EntryAdapter<T, TextEntry> {
+  
+  @Override
+  public T fromEntry(final TextEntry entry) {
+    return JsonSerialization.deserialized(entry.entryData(), entry.typed());
+  }
+
+  @Override
+  public TextEntry toEntry(final T source, final Metadata metadata) {
+    final String serialization = JsonSerialization.serialized(source);
+    return new TextEntry(PortfolioCreated.class, 1, serialization, metadata);
+  }
+
+  @Override
+  public TextEntry toEntry(final T source, final String id, final Metadata metadata) {
+    final String serialization = JsonSerialization.serialized(source);
+    return new TextEntry(id, PortfolioCreated.class, 1, serialization, metadata);
```

Comment:
This should work for the first event, but if we want to add more I would change it:

```suggestion
    return new TextEntry(id, source.getClass(), 1, serialization, metadata);
```